crates/languages/src/go/runnables.scm

```diff
@@ -3,25 +3,52 @@
     (function_declaration name: (_) @run @_name
         (#match? @_name "^Test.+"))
         (#set! tag go-test)
-)
+) @go-test
 
 ; `t.Run`
 (
-    (call_expression function: (_) @run @_name
-        (#match? @_name "^t.Run.*"))
-        (#set! tag go-subtest)
-)
+    (call_expression 
+      function: (
+      	selector_expression
+      	field: _ @run @_name
+        (#eq? @_name "Run")
+      )
+      arguments: (
+        argument_list
+        . 
+        (interpreted_string_literal) @_subtest_name
+        .
+        (func_literal
+          parameters: (
+            parameter_list
+            (parameter_declaration
+              name: (identifier) @_param_name
+              type: (pointer_type
+                (qualified_type
+                  package: (package_identifier) @_pkg
+                  name: (type_identifier) @_type
+                  (#eq? @_pkg "testing")
+                  (#eq? @_type "T")
+                )
+              )
+            )
+          )
+        ) @second_argument
```

Comment:
I changed here to check second parameter based on your mentions, could you check here @osiewicz please?

Reply:
This looks good to me! We can hide the `second_argument` from user though, but that's relatively minor:
```suggestion
        (func_literal
          parameters: (
            parameter_list
            (parameter_declaration
              name: (identifier) @_param_name
              type: (pointer_type
                (qualified_type
                  package: (package_identifier) @_pkg
                  name: (type_identifier) @_type
                  (#eq? @_pkg "testing")
                  (#eq? @_type "T")
                )
              )
            )
          )
        ) @_second_argument
```